### PR TITLE
Accounts, multi sig

### PR DESCRIPTION
**Pull request by tomsmalley**
** Done **
- Types are in line (probably cleanup to do here)
- UI is okay for keys/accounts page
- Dialogs mostly function, but need style fixes

** TODO **
- Ensure migrations still work (currently untested)
- Full multisig (need UI direction, any UI for signing e.g. capability input sections need to change)
- Fixup dialog UIs for redesigns - some are stubs right now

### CODE REVIEW

File: frontend/src/Frontend/Store/V1.hs [hunk 1]
```diff
@@ -79,57 +76,52 @@ upgradeFromV0 v0 =
...
+    newNetworks = (\nets -> StoreFrontend_Network_Networks :=> Identity (V0.unNetworkMap $ runIdentity nets))
+      <$> DMap.lookup V0.StoreNetwork_Networks v0
+
     -- It's unfortunate that we don't have the key around or access to crypto here to recreate the keys.
     -- TODO: Fiddle with this so we don't need to fake out the key
     -- I don't think we should run crypto derivation functions here. The web
     -- version would generate a new key! Perhaps we should punt this to be fixed
     -- upon key restoration by the user.
-    splitOldKey (keyIdx, V0.SomeAccount_Deleted) = ([(keyIdx, Key fakeKeyPair True (mkAccountNotes ""))], mempty)
+    -- TODO this needs attention if we can't hide anymore
```
Comment (tomsmalley) [inline frontend/src/Frontend/Store/V1.hs]: This is a weird one now.

File: frontend/src/Frontend/Store/V0.hs [hunk 2]
```diff
@@ -54,7 +54,8 @@ instance ArgDict c (StoreFrontend key) where
   type ConstraintsFor (StoreFrontend key) c
     = ( c (Accounts key)
       , c PublicMeta
-      , c (Map NetworkName [NodeRef])
+      -- , c (Map NetworkName [NodeRef])
```
Comment (alexfmpe) [inline frontend/src/Frontend/Store/V0.hs]: Leftover

File: frontend/src/Frontend/UI/Dialogs/Send.hs [hunk 3]
```diff
@@ -166,94 +192,81 @@ previewTransfer model fromAccount fromGasPayer toAddress crossChainData amount =
...
           ]
         }
-      dat = case _kadenaAddress_accountCreated toAccount of
-        AccountCreated_No
-          -- This makes the assumption that the only non-created accounts are
-          -- non-vanity: that is, the name is the public key.
-          | Right pk <- parsePublicKey (unAccountName $ _kadenaAddress_accountName toAccount)
-          -> HM.singleton "key" $ Aeson.toJSON $ KeySet (Set.singleton $ toPactPublicKey pk) (Name $ BareName "keys-all" def)
+      dat = case mKeyset of
+        -- TODO check against chain
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/Send.hs]: Does this block the PR?

File: frontend/src/Frontend/UI/Dialogs/Send.hs [hunk 4]
```diff
@@ -527,13 +553,15 @@ finishCrossChainTransferConfig model fromAccount ucct = Workflow $ do
...
           chain = pure $ Just toChain
       gasAcc <- uiSenderDropdown cfg model chain never
       pure $ ffor3 (model ^. wallet_accounts) (model ^. network_selectedNetwork) gasAcc $ \netToAccount net ma -> do
-        accounts <- Map.lookup net $ unAccountStorage netToAccount
-        a <- ma
-        lookupAccountRef a accounts
+        accounts <- Map.lookup net $ unAccountData netToAccount
+        n <- ma
+        AccountInfo _ chains <- Map.lookup (fst n) accounts
+        _acc <- Map.lookup fromChain chains
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/Send.hs]: Sanity check or leftover?
Maybe use `guard`

File: frontend/src/Frontend/UI/Dialogs/Send.hs [hunk 5]
```diff
@@ -653,23 +682,25 @@ crossChainTransfer logL netInfo keys fromAccount toAccount fromGasPayer crossCha
...
-  let fromChain = accountChain fromAccount
-      toChain = either _kadenaAddress_chainId accountChain toAccount
-      toAddress = either id accountToKadenaAddress toAccount
+  let (fromName, fromChain, _) = fromAccount
+      toChain = either _kadenaAddress_chainId (view _2) toAccount
+      toAddress = either id (\(n, c, _) -> KadenaAddress n c Nothing) toAccount
   -- Client envs for making requests to each chain
   let envFromChain = mkClientEnvs nodeInfos fromChain
       envToChain = mkClientEnvs nodeInfos toChain
+  -- TODO *always* look up the guard and check the validity
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/Send.hs:692]: Does this block this PR?

File: frontend/src/Frontend/UI/Dialogs/Send.hs [hunk 6]
```diff
@@ -653,23 +682,25 @@ crossChainTransfer logL netInfo keys fromAccount toAccount fromGasPayer crossCha
...
+    Left ka -> case _kadenaAddress_keyset ka of
+      Nothing -> lookupKeySet logL networkName envToChain publicMeta ka
       -- If the account hasn't been created, don't try to lookup the guard. Just
       -- assume the account name _is_ the public key (since it must be a
       -- non-vanity account).
-      AccountCreated_No -> pure $ ffor pb $ \() -> case textToKey $ unAccountName $ _kadenaAddress_accountName ka of
-        Nothing -> Left "Couldn't make public key from name"
-        Just k -> Right $ toKS $ toPactPublicKey k
-    Right ours -> pure $ Right (toKS $ toPactPublicKey $ accountKey ours) <$ pb
+      Just ks -> pure $ Right (toPactKeyset ks) <$ pb -- TODO verify against chain
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/Send.hs:704]: Does this block this PR?

File: frontend/src/Frontend/UI/Dialogs/Send.hs (hunk 3 above)
Comment (tomsmalley) [reply frontend/src/Frontend/UI/Dialogs/Send.hs]: No, I think any of these "check against chain / lookup guard" things are just nice to have. We already will have looked it up at some point (hopefully fairly recently), but it'd be nice to ensure we are up to date.

File: frontend/src/Frontend/UI/Dialogs/Send.hs (hunk 4 above)
Comment (tomsmalley) [reply frontend/src/Frontend/UI/Dialogs/Send.hs]: Sanity check. `guard` works for me.

File: frontend/src/Frontend/Store/V1.hs (hunk 1 above)
Comment (tomsmalley) [reply frontend/src/Frontend/Store/V1.hs]: Possibly addressed by https://github.com/kadena-io/chainweaver/pull/468

File: frontend/src/Frontend/UI/Wallet.hs [hunk 7]
```diff
@@ -175,33 +177,89 @@ uiAccountItems model = do
...
-        ]
+uiAccountItem keys name accountInfo = do
+  let chainMap = _accountInfo_chains <$> accountInfo
+      notes = _accountInfo_notes <$> accountInfo
+  rec
+    (clk, dialog) <- keyRow visible notes $ sum . catMaybes <$> balances
+    visible <- toggle False clk
+    results <- listWithKey chainMap $ accountRow visible
+    let balances :: Dynamic t [Maybe AccountBalance]
+        balances = join $ traverse fst . Map.elems <$> results
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Wallet.hs]:
```suggestion
        balances = fmap Map.elems $ joinDynThroughMap $ (fmap . fmap) fst results
```

File: frontend/src/Frontend/Wallet.hs [hunk 8]
```diff
@@ -119,43 +108,51 @@ makePactLenses ''WalletCfg
...
   deriving Generic
 
+data Account = Account
+  { _account_status :: AccountStatus AccountDetails
+  , _account_storage :: VanityAccount
+  } deriving (Eq, Show)
+
+emptyAccount :: Account
+emptyAccount = Account
+  { _account_storage = VanityAccount
```
Comment (alexfmpe) [inline frontend/src/Frontend/Wallet.hs]: `blankVanityAccount`?

File: frontend/src/Frontend/Wallet.hs [hunk 9]
```diff
@@ -200,139 +180,114 @@ makeWallet
...
-      , foldr (.) id . fmap updateAccountBalance <$> newBalances
+      , foldr (.) id . fmap updateAccountStatus <$> newStatuses
       , ffor (_walletCfg_setCrossChainTransfer conf) updateCrossChain
-      , ffor (_walletCfg_delAccount conf) hideAccount
-      , ffor pubKeysDeleted hideAccountsOfKey
+      , ffor (_walletCfg_delAccount conf) removeAccount
       ]
-    let visibleAccounts = filterVisibleAccountStorage <$> accounts
+
+  performEvent_ $ liftIO . print <$> updated accounts
```
Comment (alexfmpe) [inline frontend/src/Frontend/Wallet.hs]: Intentional print?

File: frontend/src/Frontend/Wallet.hs [hunk 10]
```diff
@@ -378,16 +330,15 @@ parseWalletKeyPair errPubKey privKey = do
...
   where
-    getErr net (AccountStorage networks) mChain k = do
+    getErr net (AccountData networks) k = do
       acc <- mkAccountName k
       maybe (Right acc) (\_ -> Left "This account name is already in use") $ do
         accounts <- Map.lookup net networks
-        chains <- Map.lookup acc $ _accounts_vanity accounts
-        chain <- mChain
-        Map.lookup chain chains
+        _ <- Map.lookup acc accounts
```
Comment (alexfmpe) [inline frontend/src/Frontend/Wallet.hs]: `guard`

File: frontend/src/Pact/Server/ApiV1Client.hs [hunk 11]
```diff
@@ -1,37 +1,200 @@
...
+apiV1Client = ApiV1Client
+  { send = \appendLog batch@(SubmitBatch commands) -> do
+    url <- asks baseUrl
+    timestamp <- liftIO getCurrentTime
+    for_ commands $ \command -> liftIO $ appendLog $ CommandLog
+      { _commandLog_command = command
+      , _commandLog_timestamp = timestamp
+      , _commandLog_url = T.pack $ showBaseUrl url
+      }
+    sendF batch
```
Comment (alexfmpe) [inline frontend/src/Pact/Server/ApiV1Client.hs]: Confusing indentation

**Pull request merged**